quickwit/quickwit-actors/src/scheduler.rs

```diff
@@ -199,62 +250,103 @@ impl Scheduler {
         let timeout_event = self.timeout_event(new_evt_deadline, callback);
         self.future_events.push(Reverse(timeout_event));
         if is_new_next_deadline {
-            self.schedule_next_timeout(ctx);
+            self.schedule_next_timeout();
         }
     }
 
-    async fn process_simulate_advance_time(
-        &mut self,
-        time_shift: TimeShift,
-        tx: Sender<()>,
-        ctx: &ActorContext<Self>,
-    ) {
-        let now = self.simulated_now();
-        let deadline = match time_shift {
-            TimeShift::ToInstant(instant) => instant,
-            TimeShift::ByDuration(duration) => now + duration,
+    async fn process_simulate_sleep(&mut self, oneshot_tx: oneshot::Sender<()>, timeout: Duration) {
+        let Some(scheduler_client) = self.scheduler_client() else { return; };
+        let accelerate_time_count_guard = AccelerateTimeCountGuard::new(scheduler_client);
```

Comment:
Note: I found a bit quirky the increment/decrement dance with new/drop with the scheduler client. I might have tried something else: put the `accelerate_time_count_guard` inside the scheduler itself. The downside is that we now have a counter on the scheduler client and one on the scheduler... 
I added a couple of docs to help the reader.

Reply:
In the scheduler? Do you mean in the context maybe?